src/include/squid/complex.rb

```diff
@@ -49,11 +49,98 @@ def initialize_squid_complex(include_target)
       Yast.include include_target, "squid/store_del.rb"
       Yast.include include_target, "squid/inits.rb"
 
-
       @main_caption = _("Squid")
+      @widget_descr = {}
+      @screens = {}
+
+      load_widgets
+      load_screens
+    end
+
+    def ReallyAbort
+      !Squid.GetModified || Popup.ReallyAbort(true)
+    end
+
+    def PollAbort
+      UI.PollInput == :abort
+    end
+
+    # Read settings dialog
+    #
+    # @return [Symbol] :abort when settings could not be read; :next otherwise
+    def ReadDialog
+      Wizard.RestoreHelp(Ops.get_string(@HELPS, "read", ""))
+      Squid.AbortFunction = fun_ref(method(:PollAbort), "boolean ()")
+
+      return :abort unless Confirm.MustBeRoot
+      return :abort unless PackageSystem.CheckAndInstallPackages(["squid"])
+      return :abort unless Squid.Read
+
+      load_service_widget
+
+      :next
+    end
+
+    # Write settings dialog
+    #
+    # @return [Symbol] :abort if aborted, :next otherwise
+    def WriteDialog
+      help = @HELPS.fetch("write") { "" }
```

Comment:
`@HELPS` is defined by ourselves here, handling a possibly missing value is over engineering IMHO.
```suggestion
      help = @HELPS["write"]
```

Reply:
Actually I did not touch this code. However, it is highlighted as a diff because I move two big initializations to their own methods... 

Anyway, I'll apply your proposed changes :)